### PR TITLE
Fix HTTPS listener aggregation with kube gateway

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

HTTPS listeners on kube gateways were not aggregating properly due to translating away from privileged ports.

## Code changes
- Translate port before looking up if we already have listener

# Context

Users ran into this bug when applying a Gateway with multiple HTTPS listeners using the same port

See slack conversation [here](https://solo-io-corp.slack.com/archives/C07077NS0NP/p1720729966605369)

## Testing steps

Translation unit tests added. Can be manually verified using the resource specified in the issue.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/solo-projects/issues/6559

### CODE REVIEW

File: changelog/v1.18.0-beta8/listeners-reuse-ports.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/solo-projects/issues/6559
+    resolvesIssue: true
```
Comment (jbohanon) [inline changelog/v1.18.0-beta8/listeners-reuse-ports.yaml:4]:
```suggestion
    resolvesIssue: false
```

File: projects/gateway2/translator/testutils/inputs/multiple-listeners-http-routing/gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    hostname: httpbin.example.com
+    name: http-httpbin
+    port: 80
+    protocol: HTTP
+    tls:
+      certificateRefs:
+      - group: ""
+        kind: Secret
+        name: httpbin-tls-secret
+      mode: Terminate
```
Comment (jbohanon) [inline projects/gateway2/translator/testutils/inputs/multiple-listeners-http-routing/gateway.yaml:20]:
```suggestion
```

File: projects/gateway2/translator/testutils/inputs/multiple-listeners-http-routing/gateway.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    hostname: bookinfo.example.com
+    name: http-bookinfo
+    port: 80
+    protocol: HTTP
+    tls:
+      certificateRefs:
+      - group: ""
+        kind: Secret
+        name: bookinfo-tls-secret
+      mode: Terminate
```
Comment (jbohanon) [inline projects/gateway2/translator/testutils/inputs/multiple-listeners-http-routing/gateway.yaml:33]:
```suggestion
```

**Review by sam-heilbron — APPROVED** [review]

File: projects/gateway2/translator/gateway_translator_test.go [hunk 4]
```diff
@@ -68,6 +68,60 @@ var _ = Describe("GatewayTranslator", func() {
 		}]).To(BeTrue())
 	})
 
+	It("should translate a gateway with http routing with multiple listeners on the same port", func() {
```
Comment (jenshu) [inline projects/gateway2/translator/gateway_translator_test.go:71]: i haven't checked, but do we also have tests that check different combinations of 443 / 8443 and 8443 / 8443 get aggregated correctly?

**Review by jenshu — APPROVED** [review]

File: projects/gateway2/translator/gateway_translator_test.go (hunk 4 above)
Comment (jbohanon) [reply projects/gateway2/translator/gateway_translator_test.go:71]: No we do not have those tests, but I have validated that it works locally

**Pull request merged**